client/src/components/sources/index.tsx

```diff
@@ -128,6 +124,7 @@ const SourcesView = () => {
                               id: source.id
                             }
                           });
+                          dispatch(deselectSource());
```

Comment:
Why would a source be selected when we try to delete one?

Reply:
because we forgot to dispatch that event in many places....
I'll try to fix that.